lib/cisco_node_utils/cmd_ref/README_YAML.md

```diff
@@ -154,40 +202,59 @@ instead of the more repetitive (but equally valid):
+  get_context: '/^interface <name>$/i'
+  get_value: '/^switchport access vlan (.*)$/'
+  set_context: 'interface <name>'
+  set_value: 'switchport access vlan <number>'
 
 description:
-  config_get: 'show running-config interface all'
-  config_get_token: ['/^interface <name>$/i', '/^description (.*)$/']
-  config_set: ['interface <name>', 'description <desc>']
+  get_command: 'show running-config interface all'
+  get_context: '/^interface <name>$/i'
+  get_value: '/^description (.*)$/'
+  set_context: 'interface <name>'
+  set_value: 'description <desc>'
 
 ...
 ```
 
-### Platform and API variants
+### Data format variants
+
+Clients for different Cisco platforms may use different data formats. NXAPI (used for Cisco Nexus platforms) supports a CLI-based data format (essentially a wrapper for the Nexus CLI) as well as a NXAPI-specific structured format for some 'show' commands. Currently the gRPC client provided here (used for Cisco IOS XR platforms) supports a CLI-based format. Other platforms may have other formats such as YANG. As different formats have different requirements, the YAML must be able to accommodate this.
 
-Clients for different Cisco platforms may use different APIs. Currently these include NXAPI (CLI-based API used for Cisco Nexus platforms) and gRPC (CLI-based API used for Cisco platforms running IOS XR). Often the CLI or other input/output formats needed will vary between APIs, so the YAML must be able to accomodate this.
+CLI is the lowest common denominator, so YAML entries not otherwise flagged as applicable to a specific API type will be assumed to reference CLI. Other API types can be indicated by using the API type as a key (`cli`, `nxapi_structured`, `yang`, etc.). For example, Nexus platforms support a structured form of 'show version', while other clients might use the same command but will need to parse CLI output with a regular expression:
 
-Any of the attribute properties can be subdivided by platform and API type by using the
-combination of API type and platform type as a key. For example, interface VRF membership defaults to "" (no VRF) on both Nexus and IOS XR platforms, but the CLI is 'vrf member <vrf>' for Nexus and 'vrf <vrf>' for IOS XR. Thus, the YAML could be written as:
+```yaml
+# show_version.yaml
+description:
+  get_command: 'show version'
+  nexus:
+    data_model: nxapi_structured
+    get_value: 'chassis_id'
+  else:
+    data_model: cli
+    get_value: '/Hardware\n  cisco (([^(\n]+|\(\d+ Slot\))+\w+)/'
+```
+
+### Platform variants
+
+Even for clients using the same data format (e.g., CLI), there may be differences between classes of Cisco platform.  Any of the attribute properties can be subdivided by platform type by using the platform type as a key. For example, interface VRF membership defaults to "" (no VRF) on both Nexus and IOS XR platforms, but the CLI is 'vrf member <vrf>' for Nexus and 'vrf <vrf>' for IOS XR. Thus, the YAML could be written as:
```

Comment:
The angle brackets around `<vrf>` (two occurrences) are being eaten (treated as an unknown html tag).  Are those single-quotes supposed to be back-quotes?